### PR TITLE
Add pulsar-rest-api package

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by njam** [comment]
remember to make params lowercase :)

**Comment by tomaszdurka** [comment]
@njam Please review

In socket redis package we only create node server we add nginx wrapper in cm::services::stream. Should we add nginx in this package or maybe in cargomedia package?

**Comment by tomaszdurka** [comment]
@ppp0 please review once again and merge if ok

File: modules/pulsar_rest_api/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  content => template("${module_name}/init.sh"),
+  owner   => '0',
+  group   => '0',
+  mode    => '0755',
+  notify  => Service['pulsar-rest-api'],
+  before  => Package['pulsar-rest-api'],
+  }
+  ~>
+
+  exec { 'update-rc.d pulsar-rest-api defaults':
```
Comment (njam) [inline modules/pulsar_rest_api/manifests/init.pp]: Use `helper::service`?

**Comment by ppp0** [comment]
@njam good point.

Actually, this is important, as so far, the init script was started by `monit` as a side effect:

```
[..]
Jan 21 15:49:38 debian-7-amd64-default monit[15659]: 'pulsar-rest-api' process is not running
Jan 21 15:49:39 debian-7-amd64-default monit[15659]: 'pulsar-rest-api' trying to restart
Jan 21 15:49:39 debian-7-amd64-default monit[15659]: 'pulsar-rest-api' start: /etc/init.d/pulsar-rest-api
Jan 21 15:49:49 debian-7-amd64-default monit[15659]: 'pulsar-rest-api' process is running with pid 15680
[..]
```

**Comment by tomaszdurka** [comment]
Good to merge? Remember it's just a package and we can decide later if we should deploy on our prod servers and open to public.

**Comment by ppp0** [comment]
I think it's ok

After merging this and #832, #842 to be done

**Comment by tomaszdurka** [comment]
@njam please review

**Comment by njam** [comment]
looks good (now)

**Comment by njam** [comment]
- Remove SSL config, in favor of nginx reverse proxy?

**Comment by tomaszdurka** [comment]
@kris-lab wanna review?

**Comment by njam** [comment]
lgtm

**Comment by njam** [comment]
- Auth config doesn't work -> add test

**Comment by tomaszdurka** [comment]
- test added, it is not possible to easily test auth without providing ssl (opened issue https://github.com/cargomedia/pulsar-rest-api/issues/61)

**Comment by njam** [comment]
Only manual test then?

**Comment by njam** [comment]
looks good!

**Pull request merged**